crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -575,6 +575,155 @@ pub fn impl_reflect_value(input: TokenStream) -> TokenStream {
     )
 }
 
+#[proc_macro]
+pub fn impl_reflect_struct_and_from_reflect_struct(input: TokenStream) -> TokenStream {
+    let mut iter = input.into_iter().peekable();
+    let (r#constructor, ctor) = (iter.next().unwrap(), iter.next().unwrap());
+    match r#constructor {
+        proc_macro::TokenTree::Ident(i) => {
+            if i.to_string() != "Constructor" {
```

Comment:
I recently learned about the [`custom_keyword!`](https://docs.rs/syn/latest/syn/macro.custom_keyword.html) macro, which might be useful here and with `BevyReflectPath` (especially for letting the compiler know what should be expected).